src/overlays/actors/ovl_En_Stop_heishi/z_en_stop_heishi.c

```diff
@@ -32,45 +46,538 @@ ActorInit En_Stop_heishi_InitVars = {
+
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->unk_276 = ENSTOPHEISHI_GET_C(this);
+    this->switchFlag = ENSTOPHEISHI_GET_SWITCH_FLAG(this);
+    this->unk_288 = (this->actor.world.rot.z * 40.0f) + 50.0f;
+    this->actor.world.rot.z = 0;
+    if (this->switchFlag == 0x7F) {
+        this->switchFlag = -1;
+    }
+    if ((this->switchFlag >= 0) && Flags_GetSwitch(play, this->switchFlag)) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->actor.targetMode = 0;
+    this->actor.gravity = -3.0f;
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    this->rotYTarget = this->actor.world.rot.y;
+    this->headTurnTimer2 = 6;
+    func_80AE7E9C(this);
+}
+
+void EnStopheishi_Destroy(Actor* thisx, PlayState* play) {
+    EnStopheishi* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void EnStopHeishi_ChangeAnim(EnStopheishi* this, s32 animIndex) {
+    s32 mode;
+    f32 morphFrames;
+
+    this->currentAnim = animIndex;
+    this->currentAnimFrameCount = Animation_GetLastFrame(sAnimations[animIndex]);
+    mode = ANIMMODE_ONCE;
+    morphFrames = -10.0f;
+    if ((animIndex >= SOLDIER_ANIM_STAND_HAND_ON_HIP) && (animIndex != SOLDIER_ANIM_4)) {
+        mode = ANIMMODE_LOOP;
+    }
+    if (animIndex == SOLDIER_ANIM_5) {
+        morphFrames = 0.0f;
+    }
+    Animation_Change(&this->skelAnime, sAnimations[animIndex], 1.0f, 0.0f, this->currentAnimFrameCount, mode,
+                     morphFrames);
+}
+
+void EnStopheishi_UpdateHeadNormal(EnStopheishi* this, PlayState* play) {
+    s32 yawDiff;
+    Player* player;
+    Vec3f playerPos;
+
+    yawDiff = ABS_ALT(BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.world.rot.y));
+
+    this->headRotXTarget = 0;
+
+    if (this->actor.xzDistToPlayer < 200.0f) {
+        player = GET_PLAYER(play);
+        if (yawDiff < 0x4E20) {
+            this->headRotXTarget = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+            if (this->headRotXTarget > 0x2710) {
+                this->headRotXTarget = 0x2710;
+            } else if (this->headRotXTarget < -0x2710) {
+                this->headRotXTarget = -0x2710;
+            }
+        }
+
+        Math_Vec3f_Copy(&playerPos, &player->actor.world.pos);
+
+        if ((player->transformation == PLAYER_FORM_DEKU) || (player->transformation == PLAYER_FORM_HUMAN)) {
+            playerPos.y += 57.0f + BREG(34);
+        } else {
+            playerPos.y += 70.0f + BREG(35);
+        }
+        this->pitchToPlayer = Math_Vec3f_Pitch(&this->actor.focus.pos, &playerPos);
+    }
+}
+
+void EnStopHeishi_UpdateHeadThirdDay(EnStopheishi* this) {
+    if (this->headTurnTimer1 != 0) {
+        this->headTurnTimer1--;
+    }
+    this->pitchToPlayer = -0xBB8;
+    this->headRotXTarget = 0;
+    if (this->headTurnTimer2 < 6) {
+        this->headRotXTarget = 0x1770;
+        if (!((u32)this->headTurnTimer2 % 8)) {
```

Comment:
```suggestion
        if (((u32)this->headTurnTimer2 % 8) == 0) {
```
Should still be `== 0`,